### PR TITLE
Create mcro.md

### LINKED ISSUE #1912: Request for new ontology MCRO
<a href="https://obofoundry.org/obo-nor.github.io/dashboard/mcro/dashboard.html"></a>

### Title

Model Card Report Ontology

### Short Description

representational ontology for model card reports

### Description

Model card reports are documents detailing transparent metadata information relating to machine learning models. Similar to what we have with drug labels and nutritional labels, the goal of model cards are to communicate relevant information on all aspects of a machine learning model that have undergone any experimentation. However these important reports of the machine learning models are presented in static documents. This work encodes the structure of model card reports and align them to standard OBO Foundry ontologies to help formalize and enrich these documents. The end result is computable model of the model card that can be used to standardize reporting and be integrated in future software tooling (searching and indexing, etc.). 

### Identifier Space

mcro

### License

CC-BY 3.0

### Domain

information technology

### Source Code Repository

https://github.com/UTHealth-Ontology/MCRO

### Homepage

https://github.com/UTHealth-Ontology/MCRO

### Issue Tracker

https://github.com/UTHealth-Ontology/MCRO/issues

### Ontology Download Link

https://raw.githubusercontent.com/UTHealth-Ontology/MCRO/main/mcro.owl

### Contact Name

Tuan Amith

### Contact Email

muhammad.amith@unt.edu

### Contact GitHub Username

ProfTuan

### Contact ORCID Identifier

0000-0003-4333-1857

### Formats

- [X] OWL RDF/XML (.owl)
- [ ] OBO (.obo)
- [ ] OBO Graph JSON (.json)

### Dependencies

- iao
- swo
- prov-o
- skos

### Related

_No response_

### Usages

_No response_

### Intended Use Cases and/or Related Projects

_No response_

### Data Sources

_No response_

### Additional comments or remarks

_No response_

### OBO Foundry Pre-registration Checklist

- [X] I have read and understood [the registration process instructions](http://obofoundry.org/docs/Policy_for_OBO_namespace_and_associated_PURL_requests.html) and the [registration checklist](https://github.com/OBOFoundry/OBOFoundry.github.io/blob/master/docs/RegistrationChecklist.md).
- [X] There is no other ontology in the OBO Foundry which would be an appropriate place for my terms. If there were, I have contacted the editors, and we decided in mutual agreement that a separate ontology is more appropriate.
- [X] My ontology has a specific release file with a version IRI and a `dc:license` annotation, serialised in RDF/XML.
- [X] I understand that term definitions, while not mandatory, are key to understanding the intentions of a term especially when the ontology is used in curation. I made sure that a reasonable majority of terms in my ontology have definitions, using the [IAO:0000115](http://www.ontobee.org/ontology/IAO?iri=http://purl.obolibrary.org/obo/IAO_0000115) property.
- [X] For every term in my ontology, I checked whether another OBO Foundry ontology has one with the same meaning. If so, I re-used that term directly (not by cross-reference, by directly using the IRI).
- [X] For all relationship properties (Object and Data Property), I checked whether [the Relation Ontology (RO)](http://www.ontobee.org/ontology/catalog/RO?iri=http://www.w3.org/2002/07/owl%23ObjectProperty) includes an appropriate one. I understand that aligning with RO is an essential part of the overall alignment between OBO ontologies!
- [X] For the selection of appropriate annotation properties, I looked at [OMO](http://www.ontobee.org/ontology/catalog/OMO?iri=http://www.w3.org/2002/07/owl%23AnnotationProperty) first. I understand that aligning ontology metadata and term-level metadata is essential for cross-integration of OBO ontologies.
- [X] If I was not sure about the meaning of any of the checkboxes above, I have consulted with a member of the OBO Foundry for advice, e.g., through the [obo-discuss](https://groups.google.com/g/obo-discuss) Google Group.
- [X] The requested ID space does not conflict with another ID space found in other registries such as the Bioregistry and BioPortal, see [here](https://obofoundry.org/id-policy) for a complete list.

### PR DESCRIPTION
**Pull request by ProfTuan**
Resolves #1912

### CODE REVIEW

**Comment by jamesaoverton** [comment]
One of our automated tests failed: For your contact person, we require a GitHub username in the `github` field.

**Comment by jamesaoverton** [comment]
Thanks for adding the `github` field. I also shortened the `title` to just "Model Card Report Ontology" (to match all the other OBO projects). Is this OK with you?

There are some other problems, hopefully small, that I can't figure out. I'll leave that to @pfabry and @matentzn.

**Comment by ProfTuan** [comment]
@jamesaoverton  that's fine. Thanks!

**Comment by deepakunni3** [comment]
It looks like the error in the test is due to a difference in the formatting (observed vs expected). It would be useful to run `tox -e lint` and commit the changes for `mcro.md`. This should fix the tests.

File: ontology/mcro.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+domain: information technology
+homepage: https://github.com/UTHealth-Ontology/MCRO
+license:
+  label: CC BY 4.0
+  url: https://creativecommons.org/licenses/by/4.0/
+preferredPrefix: MCRO
+products:
+- id: mcro.owl
+repository: https://github.com/UTHealth-Ontology/MCRO
+tracker: https://github.com/UTHealth-Ontology/MCRO
```
Comment (matentzn) [inline ontology/mcro.md:23]:
```suggestion
tracker: https://github.com/UTHealth-Ontology/MCRO/issues
```

**Review by matentzn — APPROVED** [review]

File: ontology/mcro.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+title: Model Card Report Ontology
+contact:
+  email: muamith@utmb.edu
+  github: ProfTuan
+  label: Tuan Amith
+  orcid: 0000-0003-4333-1857
+dependencies:
+- id: iao
+- id: swo
+description: Model card reports are documents detailing transparent metadata information relating to machine learning models. Similar to what we have with drug labels and nutritional labels, the goal of model cards are to communicate relevant information on all aspects of a machine learning model that have undergone any experimentation. However these important reports of the machine learning models are presented in static documents. This work encodes the structure of model card reports and align them to standard OBO Foundry ontologies to help formalize and enrich these documents. The end result is computable model of the model card that can be used to standardize reporting and be integrated in future software tooling (searching and indexing, etc.).
```
Comment (cthoyt) [inline ontology/mcro.md:13]: it doesn't make sense that the description line is longer than the long text, can this be summarized in a single sentence, and the rest added in the main body of the markdown file?

**Review by cthoyt — CHANGES_REQUESTED** [review]
There are several parts in the dashboard check that are not passing: https://obofoundry.org/obo-nor.github.io/dashboard/mcro/dashboard.html

what's the policy on accepting ontologies that are not passing parts (besides the plurality of users check)?

**Comment by matentzn** [comment]
The definition of "passing" is to not have any fails, not to have only passes, so MCRO passes. 

However, @ProfTuan, 	

```
Version IRI 'https://raw.githubusercontent.com/UTHealth-Ontology/MCRO/main/mcro.owl' is not in recommended format
```

- [ ] Is a really central error, even though the dashboard is a bit lenient about it. Can you fix the version IRI in your ontology please to adhere to the [versioning principle](https://obofoundry.org/principles/fp-004-versioning.html)?

File: ontology/mcro.md [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+title: Model Card Report Ontology
+contact:
+  email: muamith@utmb.edu
+  github: ProfTuan
+  label: Tuan Amith
+  orcid: 0000-0003-4333-1857
+dependencies:
+- id: iao
+- id: swo
+description: An ontology representing the structure of model card reports to describe machine learning assests.
```
Comment (cthoyt) [inline ontology/mcro.md:13]: this still doesn't say what a model card report is

File: ontology/mcro.md [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+  url: https://creativecommons.org/licenses/by/4.0/
+preferredPrefix: MCRO
+products:
+- id: mcro.owl
+repository: https://github.com/UTHealth-Ontology/MCRO
+tracker: https://github.com/UTHealth-Ontology/MCRO/issues
+activity_status: active
+---
+
+An OWL2-encoded ontology modeling the model card report for machine learning model assests. The aim of this work is to describe machine learning models to communicate information about specific details about the model (trade offs, intended users, licensing, etc.). As an ontology-based artifact we can semantically express, standardize, and yield computable benefits to the model card reports.
```
[the remark this answers is not in the document]
Comment (cthoyt) [reply ontology/mcro.md:27]: There's plenty of text and discussion in https://github.com/OBOFoundry/OBOFoundry.github.io/issues/1912, I would expect all of it gets summarized in a thoughtful way here

**Comment by ProfTuan** [comment]
> @pfabry @cthoyt - Let's stick with the process.
> 
> 1. Errors are reported on the ontologies issue tracker, not here.
> 2. This PR has nothing to do with the ontologies acceptance, this is in the past. If you see glaring issues, open an issue here on the issue tracker and request a certain QC check to become mandatory. We vote on it, add it to the Dashboard, and next time it will be caught.
> 3. The PR is purely about ensuring that the ontologies description is sufficient.
> 
> @ProfTuan I agree though with @cthoyt that your description could be a bit more detailed, given that the main consumers of the registry will not know what a model card report is. Is a rough approximation:
> 
> 1. A single clear sentence in the description field
> 2. At least a paragraph describing the ontology and its intended uses in the main part in the YAML below. As @cthoyt suggests, you can draw from the examples in the acceptance discussion.

@matentzn @cthoyt I provided an update to .md file. However. I want to be sure, is there a description section for the .yml file? I've looked at various .yml files for obo and none of them have a description section.

**Comment by cthoyt** [comment]
It's true, we've had difficulty convincing new ontology requesters to write nice, detailed descriptions of what their ontology does, what was the motivation, etc. in the actual submission markdown, even given the fact that they already had to write it for the submission.

Here's an example of a well-written ontology description: https://obofoundry.org/ontology/mondo. In https://raw.githubusercontent.com/OBOFoundry/OBOFoundry.github.io/master/ontology/mondo.md you can clearly see where the detailed description is written. I highly encourage you to start adding the following text, which I have merely copy-pasted from your original issue:

```markdown
Model card reports are documents detailing transparent metadata information relating to machine learning models. Similar to what we have with drug labels and nutritional labels, the goal of model cards are to communicate relevant information on all aspects of a machine learning model that have undergone any experimentation. However these important reports of the machine learning models are presented in static documents. This work encodes the structure of model card reports and align them to standard OBO Foundry ontologies to help formalize and enrich these documents. The end result is computable model of the model card that can be used to standardize reporting and be integrated in future software tooling (searching and indexing, etc.).

The project was started when we noticed that NIH had mentioned an interest in model card reports for AI-based machine learning models in their bridge2AI initiative. So while the ontology resource can be used for application purposes, it serves as way to standardize and formalize model card reporting documents. The comments about the performance metric subclasses are actually document sections not the actual metric. We could import STATO later and link the metric subsections (e.g. "accuracy section" > is about > stato:accuracy). Overall the ontology just represents report documents.

## Use Case

I'm trying to understand what your use case of the ontology is - looking at the use of data properties, are you building an ontology that is used as a data model, perhaps for validation? If so, have you considered technologies like SHACL instead that might allow more complex shapes to be built for validation purposes?

I think looking at the data properites is misleading. We are not trying to validate data, as so much just to provide a template for model card reports using an ontology-based framework to formalize these types of documents for biomedical informatics research that utilize AI-based machine learning. I also feel that use case question might be a bit unfair. Without naming specific projects there are one or two ontologies that are currently on OBO which are clearly application ontologies.

We do have a paper that might clarify the scope and use case:

Amith, M.T., Cui, L., Zhi, D. et al. Toward a standard formal semantic representation of the model card report. BMC Bioinformatics 23 (Suppl 6), 281 (2022). https://doi.org/10.1186/s12859-022-04797-6
```


^ all of this text was already written. Please please please **edit AND revise** it so anyone who looks at the page for your ontology can better understand what it is, why it exists, and why it's part of obo (in a way that's nice english prose, not copy/pasted offhand comments)

**Comment by ProfTuan** [comment]
I'll revise it. I should mention that the examples and templates that was shown to me had very simple descriptions which is what I thought was expected.

**Comment by matentzn** [comment]
@ProfTuan just a quick ping so we can start listing your ontology on our site!

**Comment by matentzn** [comment]
## Remaining action items:

@ProfTuan thanks for the edits. This is better, but 

- [ ] it does not address @cthoyt comments in https://github.com/OBOFoundry/OBOFoundry.github.io/pull/2368#issuecomment-1553075382
- [ ] your version IRI is still wrong: https://github.com/OBOFoundry/OBOFoundry.github.io/pull/2368#issuecomment-1551474864

**Comment by ProfTuan** [comment]
@matentzn both are addressed to the best of my knowledge.

**Comment by cthoyt** [comment]
@deepakunni3 FYI my concerns are explicitly not yet addressed

**Comment by matentzn** [comment]
My concern is addressed, I checked it off. @ProfTuan sorry being a bit insistent, but we need to get this right now, else it will never be fixed. Can you carefully re-read @cthoyt comments above and extend your explanation according to his suggestions?

**Comment by ProfTuan** [comment]
@matentzn I added what I thought was missing along what @deepakunni3 had mentioned. Let me know if there's any issues.

File: ontology/mcro.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    url: https://github.com/UTHealth-Ontology/MCRO-Software
+  publications: 
+  - id: https://doi.org/10.1145/3543873.3587601
+    title: 'Application of an ontology for model cards to generate computable artifacts for linking machine learning information from biomedical research'
+activity_status: active
+---
+
+Model card reports are documents detailing transparent metadata information relating to machine learning models. Analogous with drug labels and nutritional labels, the goal of model cards are to communicate relevant information on all aspects of a machine learning model that have undergone any experimentation. The National Institutes of Health, through their Bridge2AI initiative, expressed interest in model cards as an artifact to communicate specifics and promote transparency for AI-based machine learning models in biomedical research. However, these reports are presented in static documents, and have the potential to impede any possible retrieval, analysis, and aggregation of these reports. Model card reports translated as RDF-based formats could help in this effort, including supplementing application tools to analyze AI-based machine learning models.
```
Comment (cthoyt) [inline ontology/mcro.md:39]: I now understand that a model card is some kind of metadata about a machine learning model, but I still don't understand in any detail.  Can you provide a detailed, concrete example of a model card (in the markdown)? Maybe an image would be helpful.

@proftuan Do you understand the issue I am having? I am a person with no knowledge of this topic and I want to understand it, so I need to be taught in a way that's written to help me learn. Other people who are reading through OBO Foundry will probably also feel the same way.

**Comment by cmungall** [comment]
I recommend sourcing the original publication my Mitchell et al
https://arxiv.org/abs/1810.03993

**Comment by handemcginty** [comment]
> Hi @ProfTuan, I saw you made an update to the PR, but I am not sure if you saw my message at [#2368 (comment)](https://github.com/OBOFoundry/OBOFoundry.github.io/pull/2368#discussion_r1224517023). Here's what it said:
> 
> > I now understand that a model card is some kind of metadata about a machine learning model, but I still don't understand in any detail. Can you provide a detailed, concrete example of a model card (in the markdown)? Maybe an image would be helpful.
> > @ProfTuan Do you understand the issue I am having? I am a person with no knowledge of this topic and I want to understand it, so I need to be taught in a way that's written to help me learn. Other people who are reading through OBO Foundry will probably also feel the same way.
> 
> Can you please comment on this? I think Chris's suggestion is good to include relevant references, but not sufficient to address my concerns. I would appreciate if you could give some feedback, because I'm not sure if you understand what I'm asking or not.

@cthoyt The paper @cmungall linked contains images of cards and details. Why wouldn't giving the paper as an example reference won't be sufficient?

**Comment by cthoyt** [comment]
@handemcginty I disagree. Expecting someone to follow a link to read and understand paper is a pretty big burden for someone reading through OBO Foundry who wants to just understand what any given ontology is about. I admit that in many places, the topics are pretty common knowledge and asking for a detailed description wasn't necessary. However, here, this topic is outside of the typical scope.

@ProfTuan it would be really important that you can let us know that you have read some of this feedback

**Comment by handemcginty** [comment]
> @handemcginty I disagree. Expecting someone to follow a link to read and understand paper is a pretty big burden for someone reading through OBO Foundry who wants to just understand what any given ontology is about. I admit that in many places, the topics are pretty common knowledge and asking for a detailed description wasn't necessary. However, here, this topic is outside of the typical scope. If the paper has nice examples, perhaps we could adapt them directly into the `mcro.md` file.
> 
> @ProfTuan it would be really important that you can let us know that you have read some of this feedback

@cthoyt I don't understand what you disagree with. I was asking why you think it is not sufficient to have the paper for this ontology is not sufficient. I see similarly situated ontologies: https://obofoundry.org/ontology/ontoavida.html and https://obofoundry.org/ontology/kisao.html citing references for further information.

**Comment by ProfTuan** [comment]
> @handemcginty I disagree. Expecting someone to follow a link to read and understand paper is a pretty big burden for someone reading through OBO Foundry who wants to just understand what any given ontology is about. I admit that in many places, the topics are pretty common knowledge and asking for a detailed description wasn't necessary. However, here, this topic is outside of the typical scope. If the paper has nice examples, perhaps we could adapt them directly into the `mcro.md` file.
> 
> @ProfTuan it would be really important that you can let us know that you have read some of this feedback

Actually I don't understand what the misunderstanding is, especially since we are referring to a short document (1-2 pages) that describe information about a thing. I did add some updates recently but what more is there to add, short of writing a pedantic description about a simple report?

File: ontology/mcro.md (hunk 4 above)
[the remark this answers is not in the document]
Comment (matentzn) [reply ontology/mcro.md]: I am also wondering at the lack of discussion here, but I am calling it now! Sorry @cthoyt - remember, win 51%, lose 49%, is a great target for success!

**Pull request merged**